qutebrowser/config/configfiles.py

```diff
@@ -483,6 +483,11 @@ def migrate(self) -> None:
             new_name='statusbar.show',
             true_value='never',
             false_value='always')
+        self._migrate_renamed_bool(
```

Comment:
This won't work - if `hints.uppercase` was `True` before, it would still result in an empty `hints.labels.`

Please take a look at some of the custom migrations like `_migrate_bindings_default` (and also `_migrate_renamed_bool` itself). It would need to be something similar, but using the `.upper()` value of `hints.labels`, in order to keep the old behavior.